### PR TITLE
Relax more `Sync` bounds on `Local`

### PR DESCRIPTION
**Pull request by JoJoJet**
# Objective

#5483 allows for the creation of non-`Sync` locals. However, it's not actually possible to use these types as there is a `Sync` bound on the `Deref` impls.

## Solution

Remove the unnecessary bounds.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_ecs/src/system/system_param.rs
Comment (JoJoJet) [inline crates/bevy_ecs/src/system/system_param.rs:693]: Note: above, there is an unnecessary `Sync` bound on the `Debug` implementation. I'm removing that impl entirely in #9588, so I didn't modify it to avoid conflicts.

**Comment by hymm** [comment]
could we add a test making a Local with a !Sync?

**Comment by alice-i-cecile** [comment]
> could we add a test making a Local with a !Sync?

@JoJoJet I think we should do this, and wait on a second approval :)

**Comment by JoJoJet** [comment]
I have added a test :)

**Review by alice-i-cecile — APPROVED** [review]

**Review by hymm — APPROVED** [review]

**Pull request merged**
